core/src/avm2/globals/string.rs

```diff
@@ -27,13 +40,98 @@ pub fn class_init<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implements `length` property's getter
+fn length<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Value::String(s) = this.value_of(activation.context.gc_context)? {
+            return Ok(s.encode_utf16().count().into());
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `String.charAt`
+fn char_at<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    match char_code_at(activation, this, args)? {
+        Value::Number(code) if code.is_finite() => Ok(AvmString::new(
+            activation.context.gc_context,
+            String::from_utf16_lossy(&[code as u16]),
+        )
+        .into()),
+        Value::Number(_) => Ok("".into()),
+        _ => Ok(Value::Undefined),
+    }
+}
+
+/// Implements `String.charCodeAt`
+fn char_code_at<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Value::String(s) = this.value_of(activation.context.gc_context)? {
+            let mut index = args
+                .get(0)
+                .unwrap_or(&Value::Number(0.0))
+                .coerce_to_number(activation)?;
```

Comment:
`coerce_to_i32` would be better here. This also naturally handles to NaN/infinity cases (coerces to 0).

Reply:
Thank you for your feedback!
At first, it seemed that your opinion was correct, but when I used `coerce_to_i32`, I found that it behaves differently from Flash in edge cases.

When I execute the following code, Flash Player returns an empty string, but Ruffle returns "a". It may be overflow.
```as3
"abcdef".charCodeAt(1.79e+308);
```

https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/String.html#charAt()
Refer to AS3 document; the document describes String.charAt, charCodeAt is function takes a Number argument. So I think that using `coerce_to_number` instead of `coerce_to_i32` seems to be the right thing to do.

Reply:
Oops, you're right. This behavior is different than AVM1.

However, seems like this code incorrectly traces `65`:
`trace("ABC".charCodeAt(Infinity));`
The `is_finite()` check causes this to change to index 0.

Reply:
Thanks for your review.
I changed the `is_finite()` to `is_nan()` check.